src/realm/object-store/c_api/error.hpp

```diff
@@ -46,9 +46,13 @@ class ErrorStorage {
     bool get_as_realm_error_t(realm_error_t* out) const noexcept;
     bool clear() noexcept;
 
+    void set_usercode_error(void* usercode_error);
+    void* get_and_clear_usercode_error();
+
 private:
     util::Optional<realm_error_t> m_err;
     std::string m_message_buf;
+    util::Optional<void*> m_usercode_error;
```

Comment:
What is the idea behind using util::Optional here? You can just use the raw pointer. You can't see the difference from the outside. Both if it is not set or set to nullptr, you will return nullptr.

Reply:
I can see the point if you change `get_and_clear_usercode_error` to return an util::Optional<void*>